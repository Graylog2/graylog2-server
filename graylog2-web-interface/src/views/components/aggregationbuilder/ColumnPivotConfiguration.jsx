/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import React from 'react';
import PropTypes from 'prop-types';

import { Button } from 'components/graylog';
import Input from 'components/bootstrap/Input';

const ColumnPivotConfiguration = ({ onClose, onRollupChange, rollup }) => (
  <div>
    <Input type="checkbox"
           id="rollup"
           name="rollup"
           label="Rollup"
           autoFocus
           onChange={(e) => onRollupChange(e.target.checked)}
           help="When rollup is enabled, an additional trace totalling individual subtraces will be included."
           checked={rollup} />
    <div className="pull-right" style={{ marginBottom: '10px' }}>
      <Button bsStyle="success" onClick={onClose}>Done</Button>
    </div>
  </div>
);

ColumnPivotConfiguration.propTypes = {
  onClose: PropTypes.func,
  onRollupChange: PropTypes.func,
  rollup: PropTypes.bool,
};

ColumnPivotConfiguration.defaultProps = {
  onClose: () => {},
  onRollupChange: () => {},
  rollup: true,
};

export default ColumnPivotConfiguration;
