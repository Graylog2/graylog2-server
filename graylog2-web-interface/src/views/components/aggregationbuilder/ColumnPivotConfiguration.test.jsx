import React from 'react';
import { mount } from 'wrappedEnzyme';
import 'helpers/mocking/react-dom_mock';

import ColumnPivotConfiguration from './ColumnPivotConfiguration';

describe('<ColumnPivotConfiguraiton />', () => {
  it('should render as expected', () => {
    const wrapper = mount(<ColumnPivotConfiguration rollup />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change config when clicked on checkbox', () => {
    const changeFn = jest.fn((checked) => {
      expect(checked).toBe(true);
    });
    const wrapper = mount(<ColumnPivotConfiguration onRollupChange={changeFn}
                                                    rollup={false} />);
    wrapper.find('input[name="rollup"]').simulate('change', { target: { checked: true } });
    expect(changeFn.mock.calls.length).toBe(1);
  });
});
