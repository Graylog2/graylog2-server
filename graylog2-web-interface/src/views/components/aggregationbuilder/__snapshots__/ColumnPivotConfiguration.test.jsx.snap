// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColumnPivotConfiguraiton /> should render as expected 1`] = `
<ColumnPivotConfiguration
  onClose={[Function]}
  onRollupChange={[Function]}
  rollup={true}
>
  <div>
    <Input
      addonAfter={null}
      autoFocus={true}
      bsStyle={null}
      buttonAfter={null}
      checked={true}
      help="When rollup is enabled, an additional trace totalling individual subtraces will be included."
      id="rollup"
      label="Rollup"
      name="rollup"
      onChange={[Function]}
      placeholder=""
      type="checkbox"
    >
      <Component
        controlId="rollup"
        validationState={null}
      >
        <FormGroup__StyledFormGroup
          controlId="rollup"
          validationState={null}
        >
          <StyledComponent
            controlId="rollup"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "SIZES": Array [
                  "large",
                  "small",
                ],
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "FormGroup__StyledFormGroup-sc-1wv4cm9-0",
                  "isStatic": false,
                  "lastClassName": "ekazNr",
                  "rules": Array [
                    [Function],
                    ";",
                  ],
                },
                "displayName": "FormGroup__StyledFormGroup",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "FormGroup__StyledFormGroup-sc-1wv4cm9-0",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            validationState={null}
          >
            <FormGroup
              bsClass="form-group"
              className="FormGroup__StyledFormGroup-sc-1wv4cm9-0 ekazNr"
              controlId="rollup"
              validationState={null}
            >
              <div
                className="FormGroup__StyledFormGroup-sc-1wv4cm9-0 ekazNr form-group"
              >
                <InputWrapper>
                  <span>
                    <Checkbox
                      autoFocus={true}
                      bsClass="checkbox"
                      checked={true}
                      disabled={false}
                      inline={false}
                      inputRef={[Function]}
                      label="Rollup"
                      name="rollup"
                      onChange={[Function]}
                      placeholder=""
                      title=""
                      type="checkbox"
                    >
                      <div
                        className="checkbox"
                      >
                        <label
                          title=""
                        >
                          <input
                            autoFocus={true}
                            checked={true}
                            disabled={false}
                            label="Rollup"
                            name="rollup"
                            onChange={[Function]}
                            placeholder=""
                            type="checkbox"
                          />
                          Rollup
                        </label>
                      </div>
                    </Checkbox>
                    <HelpBlock>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "HelpBlock-i8is92-0",
                              "isStatic": true,
                              "lastClassName": "huFHQV",
                              "rules": Array [
                                "display:block;margin-top:5px;margin-bottom:10px;color:",
                                "#9f9f9f",
                                ";",
                              ],
                            },
                            "displayName": "HelpBlock",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "HelpBlock-i8is92-0",
                            "target": [Function],
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <HelpBlock
                          bsClass="help-block"
                          className="HelpBlock-i8is92-0 huFHQV"
                        >
                          <span
                            className="HelpBlock-i8is92-0 huFHQV help-block"
                          >
                            When rollup is enabled, an additional trace totalling individual subtraces will be included.
                          </span>
                        </HelpBlock>
                      </StyledComponent>
                    </HelpBlock>
                  </span>
                </InputWrapper>
              </div>
            </FormGroup>
          </StyledComponent>
        </FormGroup__StyledFormGroup>
      </Component>
    </Input>
    <div
      className="pull-right"
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      <ForwardRef
        bsStyle="success"
        onClick={[Function]}
      >
        <Button
          bsStyle="success"
          onClick={[Function]}
        >
          <StyledComponent
            bsStyle="success"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "SIZES": Array [
                  "large",
                  "small",
                  "xsmall",
                ],
                "STYLES": Array [
                  "success",
                  "warning",
                  "danger",
                  "info",
                  "default",
                  "primary",
                  "link",
                ],
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Button-c9cbmb-0",
                  "isStatic": false,
                  "lastClassName": "jKsGyd",
                  "rules": Array [
                    [Function],
                  ],
                },
                "displayName": "Button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Button-c9cbmb-0",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="success"
              className="Button-c9cbmb-0 jKsGyd"
              disabled={false}
              onClick={[Function]}
            >
              <button
                className="Button-c9cbmb-0 jKsGyd btn btn-success"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Done
              </button>
            </Button>
          </StyledComponent>
        </Button>
      </ForwardRef>
    </div>
  </div>
</ColumnPivotConfiguration>
`;
